platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/platformstatus/statuslogic/ActiveStatusLogic.java

```diff
@@ -0,0 +1,89 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.system.platformstatus.statuslogic;
+
+import com.swirlds.common.system.platformstatus.PlatformStatus;
+import com.swirlds.common.system.platformstatus.PlatformStatusAction;
+import com.swirlds.common.system.platformstatus.PlatformStatusConfig;
+import com.swirlds.common.time.Time;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.time.Duration;
+import java.time.Instant;
+
+/**
+ * Class containing the state machine logic for the {@link PlatformStatus#ACTIVE ACTIVE} status.
+ */
+public class ActiveStatusLogic extends AbstractStatusLogic {
+    /**
+     * The last time an own event was observed reaching consensus
+     */
+    private Instant lastTimeOwnEventReachedConsensus;
+
+    /**
+     * Constructor
+     *
+     * @param time   a source of time
+     * @param config the platform status config
+     */
+    public ActiveStatusLogic(@NonNull final Time time, @NonNull final PlatformStatusConfig config) {
+        super(time, config);
+
+        // an own event had to reach consensus to arrive at the ACTIVE status
+        this.lastTimeOwnEventReachedConsensus = time.now();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @NonNull
+    @Override
+    public PlatformStatus processStatusAction(@NonNull final PlatformStatusAction action) {
+        return switch (action) {
+            case OWN_EVENT_REACHED_CONSENSUS -> {
+                // record the time an own event reached consensus, resetting the timer that would trigger a transition
+                // to CHECKING
+                lastTimeOwnEventReachedConsensus = getTime().now();
+                yield getStatus();
+            }
+            case FREEZE_PERIOD_ENTERED -> PlatformStatus.FREEZING;
+            case FALLEN_BEHIND -> PlatformStatus.BEHIND;
+            case STATE_WRITTEN_TO_DISK -> getStatus();
+            case CATASTROPHIC_FAILURE -> PlatformStatus.CATASTROPHIC_FAILURE;
+            case TIME_ELAPSED -> {
+                if (Duration.between(lastTimeOwnEventReachedConsensus, getTime().now())
+                                .compareTo(getConfig().activeStatusDelay())
+                        > 0) {
+                    // if an own event hasn't been observed reaching consensus in the configured duration, go back to
+                    // CHECKING
+                    yield PlatformStatus.CHECKING;
+                } else {
+                    yield getStatus();
+                }
+            }
+            default -> throw new IllegalArgumentException(getUnexpectedStatusActionLog(action));
+        };
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @NonNull
+    @Override
+    public PlatformStatus getStatus() {
+        return PlatformStatus.ACTIVE;
+    }
```

Comment:
This could be implemented in the abstract class, you would just have to pass the status into the super constructor.

Reply:
super class has been removed in latest refactor

Reply:
After I added the ability to pass data with the actions, the `time` stuff that the base class was handling before fit better elsewhere. This left only the `config` member variable, and I didn't think that warranted the added complexity of a base class.